### ISSUE TITLE
Regression in lxc-attach -s USER

### ISSUE THREAD

**Issue by overlayfs** [lxc/lxc#1890]
**Regression**
`lxc-attach -s USER`
attaches to all of a container's namespaces.  (It should only attach to the user-namespace).

**Steps to reproduce**
```
lxc-create -n test -t download -- -d ubuntu -r trusty -a amd64 # unprivileged
lxc-start -n test
lxc-attach -n test -s USER -- ls
```
This lists files in the container's mount-namespace.  (It should list files in the host's mount-namespace).

**OS details**
-lxc-daily ppa
-host & container: ubuntu-14.04 64bit
-kernel: linux-generic-lts-xenial

The regression is not present in lxc version 2.1.1+master\~20171020-0945-0ubuntu1\~trusty .  It is present in version 2.1.1+master\~20171030-1827-0ubuntu1\~trusty .

**Comment by brauner** [comment]
Ah right, I need to tweak my logic for `lxc_attach_to_ns()` in `attach.c`.

**Comment by brauner** [comment]
@overlayfs, send a patch in #1893 in case you want to test right now.

**Comment by overlayfs** [comment]
@brauner, I have tested the patch in #1893 and confirm that it fixes issue #1890.

Thank you Christian!
